### DISCUSSION TITLE
Custom record action questions

### DISCUSSION THREAD

**Discussion by jszobody** [filamentphp/filament#273] [Help] [answered]
I'm trying to build a custom record action. Specifically, I want to add an icon to impersonate users. I'm designing it along the lines of https://github.com/KABBOUCHI/nova-impersonate.

I have a few questions.

## 1) Adding the record action to the resource table definition

I've created an `Impersonate` action class with a corresponding view, and I'm trying to now add them to my `UserResource` table definition. It appears that the only way to do this is with the `->recordActions([...])` method. However this overwrites and existing record actions, removing the "Edit" link from the table row. So instead I'm doing this:

```
return $table
    ->columns([
        // ...
    ])
    ->recordActions(array_merge([
        Impersonate::make()
    ], $table->getRecordActions()));
```

Am I missing a better way to handle this? 

I feel like the `Table` class could use a `->action` or `addAction` or perhaps `prependAction` method, taking a single action and adding/prepending to the existing actions array. Thoughts?

## 2) Record action UI

This is what it looks like once added:

<img width="318" alt="CleanShot 2021-03-22 at 19 57 59@2x" src="https://user-images.githubusercontent.com/203749/112072770-ecb31e80-8b48-11eb-9b13-4289ae739e72.png">

Not too bad, but it needs a bit of spacing and alignment.  I'm thinking `flex items-center justify-end space-x-2` would help polish up this cell when multiple actions are present. Thoughts?

---

I'm happy to provide PRs. I first just wanted to make sure I was on the right path, wasn't missing anything obvious, and that you'd be open to consider these changes.

Thanks for such an awesome admin tool! Excited about where this is going.

**Comment by danharrin** [comment, marked as answer]
Hey! Yes, we'd accept PRs for the styling fix and an `addRecordActions()` method that merges the existing actions array instead of replacing it.
